yt/frontends/ahf/data_structures.py

```diff
@@ -0,0 +1,176 @@
+"""
+AHF data structures
+
+
+
+"""
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2017, yt Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+import glob
+import os
+import stat
+
+import numpy as np
+
+from yt.data_objects.static_output import \
+    Dataset, \
+    ParticleFile
+from yt.geometry.particle_geometry_handler import \
+    ParticleIndex
+from yt.utilities.cosmology import \
+    Cosmology
+
+from .fields import AHFHalosFieldInfo
+
+
+class AHFHalosFile(ParticleFile):
+    def __init__(self, ds, io, filename, file_id):
+        root, _ = os.path.splitext(filename)
+        candidates = glob.glob(root + '*.AHF_halos')
+        if len(candidates) == 1:
+            filename = candidates[0]
+        else:
+            raise ValueError('Too many AHF_halos files.')
+        names = self._read_column_names(filename)
+        self.data = np.genfromtxt(filename, names=names)
```

Comment:
I'm doing this mainly because the same data will be accessed multiple times later in `IOHandlerAHFHalos`. And I am trying to avoid doing multiple IO. Also because a single halo catalog wouldn't be very large, I don't expect to see any memory issue. In the future, this might be worth doing when we have multiple chunks for a single file. If that method is preferred now, I can move the data loading to `IOHandlerAHFHalos`. But I haven't figured out a way to avoid doing multiple IO for the same data.